### PR TITLE
feat: Migrate messenger

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
Migrate `messenger.ts`, it's test and `post-message.ts` to commercial-core

### CODE REVIEW

File: src/messenger.ts [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+			errorHandler: MessengerErrorHandler,
+		) => void
+	>,
+	persistentListeners: Array<
+		(
+			register: RegisterPersistentListener,
+			errorHandler: MessengerErrorHandler,
+		) => void
+	>,
+	errorHandler: MessengerErrorHandler,
```
Comment (Jakeii) [inline src/messenger.ts:513]: I thought passing in the raven/sentry error handler would be better rather than copying it over, but not totally sure.

**Review by arelra — APPROVED** [review]
A big one!

Very minor but what do you think about putting these files into `/messenger`? The root is getting quite busy now.

**Comment by Jakeii** [comment]
> A big one!
> 
> Very minor but what do you think about putting these files into `/messenger`? The root is getting quite busy now.

Have put post-message in the folder already :), but I was thinking about how all the exports are at the root, might start getting busy there.

I'm not sure how to get some sort of alias so that `import type { RegisterListener } from '@guardian/commercial-core/messenger';` would work, as `import type { RegisterListener } from '@guardian/commercial-core/dist/esm/messenger';` feels a bit gross.
`

**Comment by arelra** [comment]
I don't personally mind the `index.ts` file being busy and exporting everything from the root path as long as it was well organised.

We could have intermediate `index.ts`s in each folder so the the root just exports everything from each subfolder.

**Pull request merged**